components/mews/actions/create-availability-block/create-availability-block.mjs

```diff
@@ -0,0 +1,203 @@
+import app from "../../mews.app.mjs";
+
+export default {
+  name: "Create Availability Block",
+  description: "Create an availability block in Mews. [See the documentation](https://mews-systems.gitbook.io/connector-api/operations/availabilityblocks#add-availability-blocks)",
+  key: "mews-create-availability-block",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    app,
+    serviceId: {
+      propDefinition: [
+        app,
+        "serviceId",
+      ],
+      description: "Unique identifier of the Service to assign block to.",
+    },
+    rateId: {
+      propDefinition: [
+        app,
+        "rateId",
+      ],
+      description: "Unique identifier of the Rate to assign block to.",
+    },
+    firstTimeUnitStartUtc: {
+      type: "string",
+      label: "First Time Unit Start (UTC)",
+      description: "Start of the time interval, expressed as the timestamp for the start of the first time unit, in UTC timezone ISO 8601 format.",
+    },
+    lastTimeUnitStartUtc: {
+      type: "string",
+      label: "Last Time Unit Start (UTC)",
+      description: "End of the time interval, expressed as the timestamp for the start of the first time unit, in UTC timezone ISO 8601 format.",
+    },
+    state: {
+      type: "string",
+      label: "State",
+      description: "State of the availability block.",
+      options: [
+        "Confirmed",
+        "Optional",
+        "Inquired",
+        "Canceled",
+      ],
+    },
+    releasedUtc: {
+      type: "string",
+      label: "Released (UTC)",
+      description: "The moment when the block and its availability is released, in UTC timezone ISO 8601 format. Takes precedence over RollingReleaseOffset.",
+      optional: true,
+    },
+    rollingReleaseOffset: {
+      type: "string",
+      label: "Rolling Release Offset",
+      description: "Exact offset from the start of availability adjustments to the moment the availability adjustment should be released, in ISO 8601 duration format. Ignored if ReleasedUtc is specified.",
+      optional: true,
+    },
+    name: {
+      type: "string",
+      label: "Name",
+      description: "The name of the block.",
+      optional: true,
+    },
+    voucherCode: {
+      type: "string",
+      label: "Voucher Code",
+      description: "Voucher code providing access to specified private Rate.",
+      optional: true,
+    },
+    bookerId: {
+      type: "string",
+      label: "Booker ID",
+      description: "Unique identifier of the Booker as a creator of an availability block.",
+      optional: true,
+    },
```

Comment:
In `update-reservation`, `bookerId` uses the propDefinition, `customerId`. Can it use those async options here?